### PR TITLE
LIVE-2755 Validator search input for ATOM and SOL

### PR DESCRIPTION
**Pull request by henrily-ledger**
### 📝 Description

Add search bar for validator list in ATOM and SOL for LLD and LLM (see screenshot)

### ❓ Context

- **Impacted projects**: `lld` `llm` `llc` 
- **Linked resource(s)**: https://ledgerhq.atlassian.net/browse/LIVE-2755 

### ✅ Checklist

- [x] **Test coverage** 
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 📸 Demo

Screenshot 2022-07-07 at 15 44 47

Screenshot 2022-07-08 at 15 23 14

Screenshot 2022-07-11 at 11 03 57

Screenshot 2022-07-11 at 11 22 42

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

File: apps/ledger-live-desktop/src/renderer/families/cosmos/DelegationFlowModal/fields/ValidatorField.jsx [hunk 1]
```diff
@@ -54,26 +59,29 @@ const ValidatorField = ({
...
-    </ValidatorsFieldContainer>
+    <>
+      {showAll && <ValidatorSearchInput noMargin={true} search={search} onSearch={onSearch} />}
+      <ValidatorsFieldContainer>
+        <Box p={1}>
+          <ScrollLoadingList
+            data={
+              showAll
+                ? validators
+                : [validators.find(v => v.validatorAddress === chosenVoteAccAddr) || validators[0]]
```
Comment (gre) [inline apps/ledger-live-desktop/src/renderer/families/cosmos/DelegationFlowModal/fields/ValidatorField.jsx:70]: calculating the data prop through a `useMemo` would help having to .find() again as well as keeping the array ref.

File: apps/ledger-live-desktop/src/renderer/families/solana/shared/fields/ValidatorsField.jsx [hunk 2]
```diff
@@ -41,12 +43,16 @@ const ValidatorField = ({ t, account, onChangeValidator, chosenVoteAccAddr, stat
...
-        validator.voteAccount.toLowerCase().includes(search)
+    return validators
+      .filter(validator => {
+        return (
+          validator.name?.toLowerCase().includes(search.toLowerCase()) ||
+          validator.voteAccount.toLowerCase().includes(search.toLowerCase())
+        );
+      })
+      .filter(
+        (value, index, self) => index === self.findIndex(t => t.voteAccount === value.voteAccount),
```
Comment (gre) [inline apps/ledger-live-desktop/src/renderer/families/solana/shared/fields/ValidatorsField.jsx:54]: feels like there is a faster algo to figure out. Here we are `O(n2)`

**Review by gre — COMMENTED** [review]
good overall but some warning on perf

File: apps/ledger-live-desktop/src/renderer/families/solana/shared/fields/ValidatorsField.jsx (hunk 2 above)
Comment (henrily-ledger) [reply apps/ledger-live-desktop/src/renderer/families/solana/shared/fields/ValidatorsField.jsx:54]: Screenshot 2022-07-12 at 13 30 01

Indeed as fun playing with the benchmark, thanks a lot !

**Review by gre — CHANGES_REQUESTED** [review]

**Review by gre — APPROVED** [review]

**Pull request merged**